src/main/java/org/sigmah/offline/dao/UserAsyncDAO.java

```diff
@@ -182,6 +185,66 @@ public void onSuccess(OrgUnitDTO result) {
         });
 	}
 
+public void getByOrgUnit(final int orgUnitId, final AsyncCallback<ListResult<UserDTO>> callback) {
+		openTransaction(Transaction.Mode.READ_ONLY, new OpenTransactionHandler() {
+
+			@Override
+			public void onTransaction(final Transaction transaction) {
+				final ObjectStore userObjectStore = transaction.getObjectStore(Store.USER);
+				final Index orgUnitIndex = userObjectStore.index("orgUnit");
+				final OpenCursorRequest openCursorRequest = orgUnitIndex.openCursor(IDBKeyRange.only(orgUnitId));
+
+				final ArrayList<UserDTO> users = new ArrayList<UserDTO>();
+
+				openCursorRequest.addCallback(new AsyncCallback<Request>() {
+
+					@Override
+					public void onFailure(Throwable caught) {
+						callback.onFailure(caught);
+					}
+
+					@Override
+					public void onSuccess(Request request) {
+						final Cursor cursor = openCursorRequest.getResult();
+						if (cursor != null) {
+							final UserJS userJS = (UserJS) cursor.getValue();
+							users.add(userJS.toDTO());
+							cursor.next();
+
+						} else {
+							callback.onSuccess(new ListResult<UserDTO>(users));
+						}
+					}
+				});
+			}
+		});
+	}
+
+	public void getByOrgUnits(final Set<Integer> orgUnitIds, final AsyncCallback<ListResult<UserDTO>> callback) {
+		final List<UserDTO> users = new ArrayList<UserDTO>();
+		// `i` needs to be final as it is accessed asynchronously
+		final int[] i = {0};
```

Comment:
Will "i" is quite a clumsy variable name :wink: 

I guess you are trying to do a closure but since you want to update the integer but you can't since it's final, you circumvented that by using a array. If i'm right, could you replace that by using an instance of `org.apache.commons.lang3.mutable.MutableInt`.

```java
final MutableInt actualSize = new MutableInt();
```

and in the callback, replace

```java
i[0]++
```

by

```java
actualSize.increment();
```